scripts/buildDocs.ts

```diff
@@ -111,24 +112,43 @@ const formatterDocumentation: IDocumentation = {
  */
 function buildDocumentation(documentation: IDocumentation) {
     // Create each module's documentation file.
-    const paths = glob.sync(documentation.globPattern);
-    const metadataJson = paths.map((path: string) => {
-        return buildSingleModuleDocumentation(documentation, path);
-    });
+    const modulePaths = glob.sync(documentation.globPattern);
+    const metadataJson = modulePaths.map((modulePath: string) =>
+        buildSingleModuleDocumentation(documentation, modulePath));
+
+    // Delete outdated directories
+    const rulesDirs = metadataJson.map((metadata: any) => metadata[documentation.nameMetadataKey]);
+    deleteOutdatedDocumentation(documentation.subDirectory, rulesDirs);
 
     // Create a data file with details of every module.
     buildDocumentationDataFile(documentation, metadataJson);
 }
 
 /**
+ * Deletes directories which are outdated
+ * @param directory Path from which outdated subdirectories have to be checked and removed
+ * @param rulesDirs The names of the current and new rules documentation directories
+ */
+function deleteOutdatedDocumentation(directory: string, rulesDirs: string[]) {
+    // find if the thing at particular location is a directory
+    const isDirectory = (source: string) => fs.lstatSync(source).isDirectory();
+    // get all subdirectories in source directory
+    const getDirectories = (source: string) => fs.readdirSync(source).filter((name) => isDirectory(path.join(source, name)));
+
+    const subDirs = getDirectories(directory);
+    const outdatedDirs = subDirs.filter((dir) => rulesDirs.indexOf(dir) < 0);
+    outdatedDirs.forEach((outdatedDir) => rimraf(path.join(directory, outdatedDir), () => {})); // tslint:disable-line:no-empty
+}
+
+/**
  * Produces documentation for a single file/module.
  */
 function buildSingleModuleDocumentation(documentation: IDocumentation, modulePath: string): Metadata {
     // Load the module.
     // tslint:disable-next-line:no-var-requires
     const module = require(modulePath);
     const DocumentedItem = module[documentation.exportName] as Documented;
-    if (DocumentedItem != null && DocumentedItem.metadata != null) {
+    if (DocumentedItem !== null && DocumentedItem.metadata !== null) {
```

Comment:
I guess you should just change it to `!= undefined` to make the linter happy and still preserve the semantics.

Reply:
That's right

Reply:
this still needs to be addressed

Reply:
Addressed in new commit